gem/middleware.py

```diff
@@ -54,6 +55,25 @@ class GemMoloGoogleAnalyticsMiddleware(MoloGoogleAnalyticsMiddleware):
     into the MoloGoogleAnalyticsMiddleware class and override
     submit_to_local_account
     '''
+    def get_article_tags(self, request):
+        """get the tags in an article if the request is for an article page"""
+        tags = []
+        path = request.get_full_path()
+        path_components = [component for component in path.split('/')
+                           if component]
+
+        try:
+            page, args, kwargs = request.site.root_page.specific.route(
+                request,
+                path_components)
+            if issubclass(type(page), ArticlePage):
+                tags = page.tags_list()
```

Comment:
Do you not need `page.specific.tags_list()`? Just wondering if this is working..

Reply:
Also, is there a chance that `page.tags_list()` will return `None`? Because that will be a problem for the join

Reply:
it will return  an empty arrayset [] when there a are no tags
`if issubclass(type(page), ArticlePage):` will ensure that it's always an ariclepage and the tags_list() will always return an empty list when there are no tags

Reply:
I think the question is whether `route()` returns a concrete type (i.e Page vs ArticlePage). The only way to be sure is to call `page.specific`

Reply:
```suggestion
                tags = page.specific.tags_list()
```